tests/core/validation/test_validation.py

```diff
@@ -0,0 +1,408 @@
+import pytest
+
+from evm.exceptions import (
+    ValidationError,
+)
+from evm.constants import (
+    SECPK1_N,
+)
+from evm.validation import (
+    validate_block_number,
+    validate_canonical_address,
+    validate_gt,
+    validate_gte,
+    validate_is_boolean,
+    validate_is_bytes,
+    validate_is_integer,
+    validate_length,
+    validate_length_lte,
+    validate_lt,
+    validate_lte,
+    validate_lt_secpk1n,
+    validate_lt_secpk1n2,
+    validate_multiple_of,
+    validate_multiple_of_8,
+    validate_stack_item,
+    validate_uint256,
+    validate_unique,
+    validate_vm_block_numbers,
+    validate_word,
+)
+
+
+@pytest.mark.parametrize(
+    "value,is_valid",
+    (
+        (tuple(), True),
+        ([], True),
+        ({}, True),
+        (set(), True),
+        ([1, 2, 3], True),
+        ([1, 2, 1], False),
+        (['a', 'b', 'c'], True),
+        (['1', '2', '3'], True),
+        (['1', '2', '1'], False),
+        (['1', '2', 1], True),
+    ),
+)
+def test_validate_unique(value, is_valid):
+    if is_valid:
+        validate_unique(value)
+    else:
+        with pytest.raises(ValidationError):
+            validate_unique(value)
+
+
+@pytest.mark.parametrize(
+    "value,is_valid",
+    (
+        (tuple(), False),
+        ([], False),
+        ({}, False),
+        (set(), False),
+        (1, False),
+        (True, False),
+        ('abc', False),
+        (b'abc', True),
+    ),
+)
+def test_validate_is_bytes(value, is_valid):
+    if is_valid:
+        validate_is_bytes(value)
+    else:
+        with pytest.raises(ValidationError):
+            validate_is_bytes(value)
+
+
+@pytest.mark.parametrize(
+    "value,is_valid",
+    (
+        (tuple(), False),
+        ([], False),
+        ({}, False),
+        (set(), False),
+        (1234, True),
+        ('abc', False),
+        # (True, False),
```

Comment:
https://stackoverflow.com/questions/2764017/is-false-0-and-true-1-in-python-an-implementation-detail-or-is-it-guarante 
According to ^ it's because in python3 booleans are a subclass of integers (t == 1, f == 0)
so in `validation.py` `isinstance(value, int)` returns `True` w/ value as a boolean

i fixed it using `if type(value) is int:` in `validation.py` but not sure if that's the best solution?